### PR TITLE
Add forgotten_room_retention_period=28d as default

### PR DESCRIPTION
**Pull request by amandahla**
### Overview

This PR adds the configuration forgotten_room_retention_period set with the value 28d. In the future, this configuration could be changed via juju config if needed.

### Rationale

Synapse creates room for DM conversations and keeps them even if are not used anymore. This wastes resources and can have an impact on performance.

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

N/A

### Checklist

- [X] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [X] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [X] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [X] The documentation is generated using `src-docs`
- [X] The documentation for charmhub is updated.
- [X] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/synapse/workload.py [hunk 1]
```diff
@@ -317,6 +317,24 @@ def enable_metrics(container: ops.Container) -> None:
...
+    Raises:
+        WorkloadError: something went wrong enabling forgotten_room_retention_period.
+    """
+    try:
+        config = container.pull(SYNAPSE_CONFIG_PATH).read()
+        current_yaml = yaml.safe_load(config)
+        current_yaml["forgotten_room_retention_period"] = "28d"
+        container.push(SYNAPSE_CONFIG_PATH, yaml.safe_dump(current_yaml))
+    except ops.pebble.PathError as exc:
+        raise WorkloadError(str(exc)) from exc
```
Comment (arturo-seijas) [inline src/synapse/workload.py:335]: Quick question, doesn't `raise WorkloadError(str(exc)) from exc` suffice?

Comment (amandahla) [reply src/synapse/workload.py:335]: If I remove it, a "error: Missing positional argument "msg" in call to "WorkloadError" happens.

File: tests/unit/test_synapse_workload.py [hunk 2]
```diff
@@ -485,6 +485,55 @@ def test_enable_metrics_error(monkeypatch: pytest.MonkeyPatch):
...
+
+    assert pull_mock.call_args[0][0] == synapse.SYNAPSE_CONFIG_PATH
+    assert push_mock.call_args[0][0] == synapse.SYNAPSE_CONFIG_PATH
+    expected_config_content = {
+        "listeners": [
+            {"type": "http", "port": 8080, "bind_addresses": ["::"]},
+        ],
+        "forgotten_room_retention_period": "28d",
+    }
+    assert push_mock.call_args[0][1] == yaml.safe_dump(expected_config_content)
```
[the remark this answers is not in the document]
Comment (amandahla) [reply tests/unit/test_synapse_workload.py]: Nice, thanks

File: src/synapse/workload.py (hunk 1 above)
Comment (arturo-seijas) [reply src/synapse/workload.py:335]: Oh, ok

**Review by arturo-seijas — APPROVED** [review]

**Review by merkata — APPROVED** [review]
lgtm

**Review by merkata — APPROVED** [review]
lgtm

**Pull request merged**